### PR TITLE
feat(get-platform): sanitise OpenSSL 3+

### PR DESCRIPTION
**Pull request by jkomyno**
This PR always returns `n.0.x` for any `n.x.x` version, where `n` is a major version with `n >= 3`.

OpenSSL 3 adopted [a new versioning policy](https://www.openssl.org/policies/general/versioning-policy.html) which is basically semver, and will not break API and ABI until OpenSSL 4, but our engines naming scheme still follows the old OpenSSL versioning scheme, where the second and not the first number indicated breaking changes.

### CODE REVIEW

File: packages/get-platform/src/platforms.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ export type Platform =
   | 'rhel-openssl-3.0.x'
   | 'linux-arm64-openssl-1.1.x'
   | 'linux-arm64-openssl-1.0.x'
-  | 'linux-arm64-openssl-3.0.x'
+  | 'linux-arm64-openssl-3.x.x'
```
Comment (aqrln) [inline packages/get-platform/src/platforms.ts:13]: I don't think we're renaming the engines just yet

Comment (jkomyno) [inline packages/get-platform/src/platforms.ts:13]:
```suggestion
  | 'linux-arm64-openssl-3.0.x'
```

File: packages/get-platform/src/platforms.ts [hunk 2]
```diff
@@ -36,7 +36,7 @@ export const platforms: Array<Platform> = [
   'rhel-openssl-3.0.x',
   'linux-arm64-openssl-1.1.x',
   'linux-arm64-openssl-1.0.x',
-  'linux-arm64-openssl-3.0.x',
+  'linux-arm64-openssl-3.x.x',
```
Comment (jkomyno) [inline packages/get-platform/src/platforms.ts:39]:
```suggestion
  'linux-arm64-openssl-3.0.x',
```

File: packages/get-platform/src/__tests__/parseOpenSSLVersion.test.ts [hunk 3]
```diff
@@ -52,6 +57,11 @@ describe('parseLibSSLVersion', () => {
       content: `/lib/libssl.so.3`,
       expect: '3.0.x',
     },
+    {
+      name: 'libssl 3.1',
+      content: `/lib/libssl.so.3.1`,
+      expect: '3.0.x',
```
Comment (Jolg42) [inline packages/get-platform/src/__tests__/parseOpenSSLVersion.test.ts:63]:
```suggestion
      expect: '3.0.x',
```
Technically `3.x.x` is better maybe but it doesn't change anything in the logic? (except that we would need to change the built engine names too on the CDN) 
So maybe not necessary to do overall?

**Review by Jolg42 — APPROVED** [review]
Looks good to me, but I probably miss some details so better if @aqrln can review too

File: packages/get-platform/src/__tests__/parseOpenSSLVersion.test.ts (hunk 3 above)
Comment (jkomyno) [reply packages/get-platform/src/__tests__/parseOpenSSLVersion.test.ts:63]: We're keeping `3.0.x` for now, which we have already tested, and revisit this decision for [Prisma 5](https://www.notion.so/prismaio/Standardise-compilation-target-names-platforms-on-Linux-a4a4c3e91d254f5ca10fcd976ea5a7da).

**Pull request merged**